controller/src/beerocks/master/son_actions.cpp

```diff
@@ -294,23 +296,18 @@ void son_actions::handle_dead_node(std::string mac, std::string hostap_mac, db &
      */
     if ((mac_type == beerocks::TYPE_IRE_BACKHAUL || mac_type == beerocks::TYPE_CLIENT) &&
         database.is_node_wireless(mac)) {
-        Socket *hostap_mac_sd = database.get_node_socket(hostap_mac);
-        if (hostap_mac_sd) {
```

Comment:
We still need this condition, but then on the MAC address:

```
if (!agent_mac.empty()) {
```

Or actually, looking at the rest, we don't do this anywhere else, so probably not needed. In that case, it should be mentoined in the commit message. Which is anyway is not entirely clear, better would be:

```
 controller: send_cmdu_to_agent: use bus instead of mrouter socket

Change "send_cmdu_to_agent" on son_actions to send the message to the
Bus instead of UDS from mrouter.

To replace the Socket, we need the master_thread as an argument to be able
to call the send_cmdu_to_bus function, and the MAC address of the agent.
Convert all callers to get the MAC address of the agent from the database 
with get_node_parent_ire instead of getting the Socket with get_node_socket.

There were four cases where the return value of get_node_socket() was
checked. Remove that check now, because we don't do it anywhere else: a
hostap node is guaranteed to have a ire parent, and if the MAC address is
empty, it will fail in the sending stage.
```

Reply:
Will change the commit message